implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -245,4 +286,38 @@ int getPriority() {
             return priority;
         }
     }
+
+    static class InterceptorWithPriority {
+        private final ConfigSourceInterceptorFactory factory;
+        private final int priority;
+
+        private InterceptorWithPriority(ConfigSourceInterceptor interceptor) {
+            this(new ConfigSourceInterceptorFactory() {
+                @Override
+                public ConfigSourceInterceptor getInterceptor(final ConfigSourceInterceptorContext context) {
+                    return interceptor;
+                }
+
+                @Override
+                public int getPriority() {
+                    return Optional.ofNullable(interceptor.getClass().getAnnotation(Priority.class))
+                            .map(Priority::value)
+                            .orElse(ConfigSourceInterceptorFactory.super.getPriority());
```

Comment:
It's unusual that the factory priority is lower than the annotation.  Doesn't that seem backwards?  As a user I would probably expect that the annotation would be used only if there was no priority given... of course that would mean that it has to be *possible* for there to be no priority.  Some ways I can think of to do that would be to define a `ConfigSourceInterceptorFactoryWithPriority` subinterface, or change `getPriority()` to return an `OptionalInt`.

Reply:
Yes, maybe we want the method to have priority :) 

Probably is better to just change it to an OptionalInt, so we don't create yet another interface.